app/javascript/hooks/request-query.ts

```diff
@@ -41,12 +42,12 @@ function handleFetch(
 }
 
 export function usePaginatedRequestQuery<TResult = unknown, TError = unknown>(
-  key: string,
+  key: QueryKey,
   request: PaginatedRequest,
   isMountedRef: React.MutableRefObject<boolean>
 ): PaginatedQueryResult<TResult, TError> {
   return usePaginatedQuery<TResult, TError>(
-    [key, request.endpoint, request.query],
+    key,
```

Comment:
I needed to change the key in order for me to be able to change the query cache easily.